d2ir/compile.go

```diff
@@ -66,6 +66,9 @@ func (c *compiler) compileSteps(m *Map) {
 		if i == 0 {
 			base = m.CopyBase(sf)
 		} else {
+			if steps.Fields[i-1].Map() == nil {
```

Comment:
Check if there's a scalar/array too and if so, error.

Reply:
None of the compiler functions return an error. You use `c.errorf()` to report it and skip the offending section.

Reply:
my main point though is that a step *must* be either a map or not set to any value.